tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_4_durability_and_cover/test_table_4_3.py

```diff
@@ -0,0 +1,179 @@
+"""Test for the concrete structural class
+according to Table 4.3 from NEN-EN 1992-1-1+C2:2011: Chapter 4 - Durability and cover to reinforcement.
+"""
+
+import pytest
+
+from blueprints.codes.eurocode.nen_en_1992_1_1_c2_2011.chapter_4_durability_and_cover.table_4_1 import (
+    Carbonation,
+    Chemical,
+    Chloride,
+    ChlorideSeawater,
+    ExposureClasses,
+    FreezeThaw,
+)
+from blueprints.codes.eurocode.nen_en_1992_1_1_c2_2011.chapter_4_durability_and_cover.table_4_3 import (
+    ConcreteMaterial,
+    ConcreteStrengthClass,
+    ConcreteStructuralClass,
+    ConcreteStructuralClassBase,
+    ConcreteStructuralClassCalculator,
+)
+from blueprints.type_alias import YEARS
+
+DUMMY_EXPOSURE_CLASSES = ExposureClasses(
+    carbonation=Carbonation.XC2,
+    chloride=Chloride.XD1,
+    chloride_seawater=ChlorideSeawater.XS1,
+    freeze=FreezeThaw.NA,
+    chemical=Chemical.XA2,
+)
+
+
+class TestConcreteStructuralClassCalculator:
+    """Unit tests for the ConcreteStructuralClassCalculator class."""
+
+    @pytest.mark.parametrize(
+        ("design_working_life", "expected_structural_class", "expected_explanation"),
+        [(50, 4, "(50 years)"), (75, 5, "(75 years)"), (100, 6, "(100 years)")],
+    )
+    def test_structural_class_delta_design_working_life(
+        self, design_working_life: YEARS, expected_structural_class: int, expected_explanation: str
+    ) -> None:
+        """Test the _structural_class_delta_design_working_life method of the ConcreteStructuralClassCalculator class."""
+        calculator = ConcreteStructuralClassCalculator(
+            DUMMY_EXPOSURE_CLASSES,
+            design_working_life,
+            ConcreteMaterial(ConcreteStrengthClass("C20/25")),
+            False,
+            False,
+        )
+
+        calculator._structural_class_delta_design_working_life()  # noqa: SLF001
```

Comment:
This feels like a workaround. The calculator shouldn't be dependent of a private interface to get the final calculations.

Reply:
It was indeed a workaround, because I was trying to test a private method.
I have changed this so the public calculate_structural_calss method will be called instead of the private method. I then check if the exact expected explanation is in the explanation. I'm also using values which would result in the default structural class, so that changing the design working life is the only factor affecting the results.

The only downside is that the private method is being tested implicitly not explicitly.